xonsh/dirstack.py

```diff
@@ -64,6 +66,13 @@ def cd(args, stdin=None):
                     return '', e.format(len(DIRSTACK))
                 else:
                     d = DIRSTACK[num - 1]
+            else:
+                # try CDPATH
+                cdpaths = env.get('CDPATH', [])
+                for cdp in cdpaths:
+                    for p in iglob(os.path.join(cdp, d)):
+                        d = p
+                        break
```

Comment:
Doesn't this only break out of the inner loop?  So you would always get the first p that is in the last cdp with a p?  I would expect to get the first p in the first cdp with a p.